inc/zoo/swar/associative_iteration.h

```diff
@@ -154,6 +154,16 @@ fullAddition(SWAR<NB, B> s1, SWAR<NB, B> s2) {
     return { result, BS{carry.value()}, BS{overflow.value()} };
 };
 
+template<int NB, typename B>
+constexpr SWAR<NB, B>
+saturatingUnsignedArithmetic(SWAR<NB, B> s1, SWAR<NB, B> s2) {
```

Comment:
`saturatingUnsignedAddition` ?

Reply:
My brother in christ you are the one that suggested s/saturatingUnsignedAddition/saturatingUnsignedArithmetic

Reply:
I think the suggestion arose in the context of a function that saturates post-operation:
```
auto saturateUnsignedArithmetic(SWAR result, ArithmeticResult flags) { // result is saturated each lane with a carry
```
`saturatingUnsignedAddition` is already a smell, since the single responsibility is not there.  I mean that the implementation of saturation does not benefit from also invoking addition, hence, it ought to be separated, the user may call addition, multiplication, exponentiation, and then saturate if desired.
But this being the first saturation operation we have implemented, we get a pass on the SRP.